HNC: e2e tests intermittent failure due to no excluded-namespace label on `kube-system` ns, etc.

After #1444 is merged on April 6, our [testgrid](https://testgrid.k8s.io/wg-multi-tenancy-hnc#periodic-e2e-tests) showed 3/10 test failure as of today.

The errors are `no endpoints available` for webhook services. It turned out that we didn't add `hnc.x-k8s.io/excluded-namespace` label to namespaces `kube-system`, `kube-public`, `kube-node-lease` before running e2e tests.
(`hnc-system` get labelled by HNC but the other 3 excluded-namespaces are required to be labelled by users.)

We need to:
* Update our `test-e2e` Makefile target to verify excluded-namespace labels first before running e2e tests and early exit if they are not properly set (just like verifying `HNC_REPAIR` value before test).
* Update testgrid to add the excluded-namespace labels first.
* Add a troubleshooting doc in HNC for "webhook service `no endpoints available`" issue that they should check if the excluded-namespace labels are added properly.

The periodic test still [failed](https://testgrid.k8s.io/wg-multi-tenancy-hnc#periodic-e2e-tests).

The error was:
```
Ensuring all system namespaces are excluded from HNC...
Error from server (InternalError): Internal error occurred: failed calling webhook "namespaces.hnc.x-k8s.io": Post "https://hnc-webhook-service.hnc-system.svc:443/validate-v1-namespace?timeout=10s": dial tcp 10.98.73.157:443: connect: connection refused
make: *** [Makefile:268: exclude-system-namespaces] Error 1
```

This is due to the VWHConfiguration is already there before we label the other system namespaces.

--- a/incubator/hnc/Makefile
+++ b/incubator/hnc/Makefile
@@ -180,12 +180,23 @@ controller-gen:
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config.
 #
-# We only delete and redeploy the deployment, and nothing else, because a)
-# deleting the CRDs will cause all the existing hierarchy configs to be wiped
-# away and b) if we don't delete the deployment, a new image won't be pulled
-# unless the tag changes.
+# We only delete the deployment and the validatingwebhookconfiguration if they
+# exist before applying the manifest, because
+# a) deleting the CRDs will cause all the existing CRs to be wiped away;
+# b) if not deleting the deployment, a new image won't be pulled unless the tag changes;
+# c) if not deleting the validatingwebhookconfiguration, we cannot label
+#    namespaces to exclude them if the HNC pod is already in a bad state.
+#
+# Then we ensure the system namespaces are excluded before we deploy HNC. This
+# step is critical because if the HNC pod is ever in a bad state, the object
+# webhook service would not respond and would stop everything in the system
+# namespaces, such as "kube-system", thus breaking the whole cluster.
 deploy: docker-push kubectl manifests
 	-kubectl -n hnc-system delete deployment hnc-controller-manager
+	-kubectl delete validatingwebhookconfiguration hnc-validating-webhook-configuration
+	-kubectl label ns kube-node-lease hnc.x-k8s.io/excluded-namespace=true --overwrite
+	-kubectl label ns kube-public hnc.x-k8s.io/excluded-namespace=true --overwrite
+	-kubectl label ns kube-system hnc.x-k8s.io/excluded-namespace=true --overwrite
 	kubectl apply -f manifests/${HNC_IMG_NAME}.yaml
 
 deploy-watch:
@@ -262,6 +273,11 @@ test-e2e-batch: exclude-system-namespaces
 		go test -v -timeout 0 ./test/e2e/... ; \
 	done
 
+# exclude-system-namespaces is called before we run any e2e tests. We do ensure
+# the system namespaces are excluded in the "deploy" target. However, we need to
+# do it here too in case users install HNC by applying manifests. Ensuring the
+# system namespaces excluded is critical, because otherwise when HNC pod is in a
+# bad state, the whole cluster will break.
 exclude-system-namespaces:
 	@echo
 	@echo "Ensuring all system namespaces are excluded from HNC..."

